tests/resources/VCH-Util.robot

```diff
@@ -84,22 +84,23 @@ Set Test Environment Variables
     Run Keyword If  ${status}  Wait Until Keyword Succeeds  5x  1s  Create Unique Bridge Network
 
 Create Unique Bridge Network
+    #Retry 3 times if ${rc} not return 0; there might be vlan conflict while parallel tests are running.
+    :FOR  ${index}  IN RANGE  3
     # Ensure unique bridges are non-overlapping in a shared build environment (our CI)
-    @{URLs}=  Split String  %{TEST_URL_ARRAY}
-    ${idx}=  Get Index From List  ${URLs}  %{TEST_URL}
-    ${lowerVLAN}=  Evaluate  (${idx}+2) * 100
-    ${upperVLAN}=  Evaluate  ${lowerVLAN}+100
+    \  @{URLs}=  Split String  %{TEST_URL_ARRAY}
+    \  ${idx}=  Get Index From List  ${URLs}  %{TEST_URL}
+    \  ${lowerVLAN}=  Evaluate  (${idx}+2) * 100
+    \  ${upperVLAN}=  Evaluate  ${lowerVLAN}+100
 
     # Set a unique bridge network for each VCH that has a random VLAN ID
-    ${vlan}=  Evaluate  str(random.randint(${lowerVLAN}, ${upperVLAN}))  modules=random
-    ${vswitch}=  Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Run  govc host.vswitch.info -json | jq -r ".Vswitch[0].Name"
-    ${rc}  ${output}=  Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Run And Return Rc And Output  govc host.portgroup.add -vlan=${vlan} -vswitch ${vswitch} VCH-%{DRONE_BUILD_NUMBER}-${vlan}
-    Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Should Be Equal As Integers  ${rc}  0
-
-    ${dvs}=  Run Keyword If  '%{HOST_TYPE}' == 'VC'  Run  govc find -type DistributedVirtualSwitch | head -n1
-    ${rc}  ${output}=  Run Keyword If  '%{HOST_TYPE}' == 'VC'  Run And Return Rc And Output  govc dvs.portgroup.add -vlan=${vlan} -dvs ${dvs} VCH-%{DRONE_BUILD_NUMBER}-${vlan}
-    Run Keyword If  '%{HOST_TYPE}' == 'VC'  Should Be Equal As Integers  ${rc}  0
-    Set Environment Variable  BRIDGE_NETWORK  VCH-%{DRONE_BUILD_NUMBER}-${vlan}
+    \  ${vlan}=  Evaluate  str(random.randint(${lowerVLAN}, ${upperVLAN}))  modules=random
+    \  ${vswitch}=  Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Run  govc host.vswitch.info -json | jq -r ".Vswitch[0].Name"
+    \  ${rc}  ${output}=  Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Run And Return Rc And Output  govc host.portgroup.add -vlan=${vlan} -vswitch ${vswitch} VCH-%{DRONE_BUILD_NUMBER}-${vlan}
+
+    \  ${dvs}=  Run Keyword If  '%{HOST_TYPE}' == 'VC'  Run  govc find -type DistributedVirtualSwitch | head -n1
+    \  ${rc}  ${output}=  Run Keyword If  '%{HOST_TYPE}' == 'VC'  Run And Return Rc And Output  govc dvs.portgroup.add -vlan=${vlan} -dvs ${dvs} VCH-%{DRONE_BUILD_NUMBER}-${vlan}
+    \  Run Keyword If  ${rc} == 0  Run Keyword And Return  Set Environment Variable  BRIDGE_NETWORK  VCH-%{DRONE_BUILD_NUMBER}-${vlan}
+    \  ...  ELSE  Log  ${output}  level=WARN
```

Comment:
I'm not sure this actually fails if it doesn't manage to create a unique bridge network.

I admit, I'd have approached this by having `Create Unique Bridge Network` call another keyword using `Wait Until Keyword Succeeds` instead of an inline loop - I think we're just attempting to recreate that same keyword function in a less obvious fashion

Reply:
Do you think that improvement is worth making? If so, we should open an issue to address this on `master` and then cherry-pick both changes together.

Reply:
We should use the 'Wait Until Keyword Succeeds', and we don't need to create another keyword, there are two places calling 'Create Unique Bridge Network'.  In VCH-util.robot, it is already implemented this way, we should change file 1-02-Docker-Pull.robot. I am going to revert the change and generate a new review.